src/query/frontend/ast/cypher_main_visitor.cpp

```diff
@@ -2336,6 +2338,94 @@ antlrcpp::Any CypherMainVisitor::visitForeach(MemgraphCypher::ForeachContext *ct
   return for_each;
 }
 
+antlrcpp::Any CypherMainVisitor::visitSchemaQuery(MemgraphCypher::SchemaQueryContext *ctx) {
+  MG_ASSERT(ctx->children.size() == 1, "SchemaQuery should have exactly one child!");
+  auto *schema_query = ctx->children[0]->accept(this).as<SchemaQuery *>();
+  query_ = schema_query;
+  return schema_query;
+}
+
+antlrcpp::Any CypherMainVisitor::visitShowSchema(MemgraphCypher::ShowSchemaContext *ctx) {
+  auto *schema_query = storage_->Create<SchemaQuery>();
+  schema_query->action_ = SchemaQuery::Action::SHOW_SCHEMA;
+  schema_query->label_ = AddLabel(ctx->labelName()->accept(this));
+  query_ = schema_query;
+  return schema_query;
+}
+
+antlrcpp::Any CypherMainVisitor::visitShowSchemas(MemgraphCypher::ShowSchemasContext * /*ctx*/) {
+  auto *schema_query = storage_->Create<SchemaQuery>();
+  schema_query->action_ = SchemaQuery::Action::SHOW_SCHEMAS;
+  query_ = schema_query;
+  return schema_query;
+}
+
+antlrcpp::Any CypherMainVisitor::visitPropertyType(MemgraphCypher::PropertyTypeContext *ctx) {
+  MG_ASSERT(ctx->symbolicName());
+  const auto property_type = utils::ToLowerCase(ctx->symbolicName()->accept(this).as<std::string>());
+  if (property_type == "bool") {
+    return common::SchemaType::BOOL;
+  }
+  if (property_type == "string") {
+    return common::SchemaType::STRING;
+  }
+  if (property_type == "integer") {
+    return common::SchemaType::INT;
+  }
+  if (property_type == "date") {
+    return common::SchemaType::DATE;
+  }
+  if (property_type == "duration") {
+    return common::SchemaType::DURATION;
+  }
+  if (property_type == "localdatetime") {
+    return common::SchemaType::LOCALDATETIME;
+  }
+  if (property_type == "localtime") {
+    return common::SchemaType::LOCALTIME;
+  }
+  throw SyntaxException("Property type must be one of the supported types!");
+}
+
+/**
+ * @return Schema*
+ */
+antlrcpp::Any CypherMainVisitor::visitSchemaTypeMap(MemgraphCypher::SchemaTypeMapContext *ctx) {
+  std::unordered_map<PropertyIx, common::SchemaType> map;
+  for (auto *property_key_pair : ctx->propertyKeyTypePair()) {
+    PropertyIx key = property_key_pair->propertyKeyName()->accept(this);
+    common::SchemaType type = property_key_pair->propertyType()->accept(this);
+    if (!map.insert({key, type}).second) {
+      throw SemanticException("Same property name can't appear twice in a schema map.");
+    }
+  }
+  return map;
+}
+
+antlrcpp::Any CypherMainVisitor::visitCreateSchema(MemgraphCypher::CreateSchemaContext *ctx) {
+  auto *schema_query = storage_->Create<SchemaQuery>();
+  schema_query->action_ = SchemaQuery::Action::CREATE_SCHEMA;
+  schema_query->label_ = AddLabel(ctx->labelName()->accept(this));
+  if (!ctx->schemaTypeMap()) {
+    throw SemanticException("Schema property map must exist!");
```

Comment:
I think this error message might not be clear for the users. What about `Schema must contain at least one property and its type!`?

Reply:
Isn't the existence of schemaTypeMap is enforced by the grammar?

Reply:
Ah yes you are right, this check is redundant